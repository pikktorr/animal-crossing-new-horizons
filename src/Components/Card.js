import React from "react";

const Card = ({ id, name, phrase }) => {
  return (
    <div className="dark-gray bg-light-yellow dib br2 pa3 ma2 shadow-5 w50 grow">
      <img
        className="br2 mw5"
        src={`http://acnhapi.com/images/villagers/${id}`}
        alt={`image_of_${name}`}
      />
      <div>
        <h2 className="f3 ma2">{name}</h2>
        <p
          className="phrase f5 ma1 gray"
          style={{ textTransform: "capitalize" }}
        >
          "{phrase}"
        </p>
      </div>
    </div>
  );
};

export default Card;
