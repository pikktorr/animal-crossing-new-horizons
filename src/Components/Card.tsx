import React from 'react';

type CardProps = {
  id: number;
  name: string;
  phrase: string;
};

const Card = ({ id, name, phrase }: CardProps) => {
  return (
    <div className="bg-yellow-200 text-slate-700 text-center rounded-xl p-4 hover:scale-105 transition duration-500 ease-in-out">
      <img
        className="rounded-md"
        src={`https://acnhapi.com/v1a/images/villagers/${id}`}
        alt={`image_of_${name}`}
      />
      <div>
        <h1 className="mt-4 uppercase font-bold">{name}</h1>
        <p
          className="italic font-light"
          style={{ textTransform: 'capitalize' }}
        >
          "{phrase}"
        </p>
      </div>
    </div>
  );
};

export default Card;
