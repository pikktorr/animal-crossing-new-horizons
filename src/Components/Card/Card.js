import React from "react";

const Card = (props) => {
  return (
    <div className="near-black bg-washed-red dib br3 pa3 ma2 grow shadow-5">
      <img
        className="br2"
        src={`http://acnhapi.com/images/villagers/${props.id}`}
        alt="character_image"
      />
      <div>
        <h2>{props.name}</h2>
        <p>"{props.phrase}"</p>
      </div>
    </div>
  );
};

export default Card;
