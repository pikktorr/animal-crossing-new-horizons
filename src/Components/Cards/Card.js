import React from "react";

const Card = ({ id, name, phrase }) => {
  return (
    <div className="dark-gray bg-washed-red dib br3 pa3 ma2 grow shadow-5 w50">
      <img
        className="br2 mw5"
        src={`http://acnhapi.com/images/villagers/${id}`}
        alt={`image_of_${name}`}
      />
      <div>
        <h2>{name}</h2>
        <p style={{ textTransform: "capitalize" }}>catchphrase: "{phrase}"</p>
      </div>
    </div>
  );
};

export default Card;
