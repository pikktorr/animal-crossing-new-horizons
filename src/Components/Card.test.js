import React from "react";
import Card from "./Card";
import { shallow } from "enzyme";

test("expect to render Card component", () => {
  expect(shallow(<Card />).debug()).toMatchSnapshot();
});
