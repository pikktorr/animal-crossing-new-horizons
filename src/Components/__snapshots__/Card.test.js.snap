// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
"<div className=\\"navy bg-light-yellow dib br2 pa3 ma2 shadow-5 w50 grow\\">
  <img className=\\"br2\\" src=\\"https://acnhapi.com/v1/images/villagers/undefined\\" alt=\\"image_of_undefined\\" loading=\\"lazy\\" />
  <div>
    <h2 className=\\"f3 ma2\\" />
    <p className=\\"phrase f5 ma1 navy\\" style={{...}}>
      &quot;
      &quot;
    </p>
  </div>
</div>"
`;
